### DISCUSSION TITLE
在Freesql中使用linq to sql 进行嵌套查询报错

### DISCUSSION THREAD

**Discussion by s899000** [dotnetcore/FreeSql#1158] [Q&A]
var t1 = (from b in
                (from a in DB.DBXAdm.Select【User】()
                 select new { a, rowid = SqlExt.RowNumber().Over().PartitionBy(a.Password).OrderBy(a.Id).ToValue() })
                 where b.rowid==1
                 select new { b.a,b.rowid }).ToList();
报 未将对象引用设置到对象的实例。 
 不加   where b.rowid==1  没报错

**Comment by 2881099** [comment]
这种浅谈查询，目前不支持，这样可以

fsql.Select\<t>()
 .Where( a=>rowid = SqlExt.RowNumber().Over().PartitionBy(a.Password).OrderBy(a.Id).ToValue() > 10 )
.ToList( a=> new { a, rowid = rowid = SqlExt.RowNumber().Over().PartitionBy(a.Password).OrderBy(a.Id).ToValue() })

**Comment by s899000** [comment]
还是不行  ，你这个写法 报 Window function is allowed only in SELECT list and ORDER BY clause”

**Comment by 2881099** [comment]
打印sql检查

**Reply by s899000** [reply]
SELECT a.`UserName` as1, a.`DeptCode` as2, a.`IsSuper` as3, a.`Password` as4, a.`RealName` as5, a.`Token` as6, a.`LastLoginTime` as7, a.`Id` as8, row_number() over( partition by a.`Password` order by a.`Id`) as9 
FROM `User` a 
WHERE (row_number() over( partition by a.`Password` order by a.`Id`) > 10)      生成的sql是这样的

**Comment by 2881099** [comment]
mysql 支持这种开窗函数吗

**Reply by s899000** [reply]
是mariadb

**Reply by s899000** [reply]
写法应该和sqlserver 一样

**Reply by 2881099** [reply]
复制到数据库执行测试就知道了
